processing/src/main/java/io/druid/segment/filter/BoundFilter.java

```diff
@@ -59,6 +61,11 @@ public boolean apply(String input)
                   if (input == null) {
                     return false;
                   }
+
+                  if (extractionFn != null) {
```

Comment:
I think it'd make more sense for DimensionPredicateFilter to take the `extractionFn` and apply it.